src/libraries/Microsoft.PowerFx.Json/Functions/IsTypeFunction_UOImpl.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    internal class IsTypeFunction_UOImpl : IsTypeFunction_UO, IAsyncTexlFunction
+    {
+        public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            var irContext = IRContext.NotInSource(FormulaType.UntypedObject);
```

Comment:
>var [](http://example.com/codeflow?start=12&length=3)

There is a lot of repetition from this function and AsTypeFunction_UOImpl; can we refactor some of the code? For example, lines 19-43 are identical